### ISSUE TITLE
Composition?

### ISSUE THREAD

**Issue by aksonov** [jbrodriguez/reactorx#2]
Great project! Is it alive :)? I don't see how to implement composition. I have several modules and don't want to mix everything together. I want something similar to combineReducers in Redux, where each reducer changes its sub-tree

**Comment by jbrodriguez** [comment]
Hi @aksonov, yes the project is alive ! :)

I'm planning to publish a v1, but I will be changing the api.

The three main concepts I want to implement are:
- Calling an action will send an object
- Saga-like middleware to handle side-effects
- Composition

This will bring the project conceptually closer to Redux, but still won't be interchangeable.

I want to keep reactorx simple to use and at the same time I want to separate store modification from side-effects (api calls, storage management, etc).

This will also improve testability.

The bottom line is that I've built projects with Redux and I definitely feel the boilerplate: mapStateToProps, mapDispatchToProps, connect and the intrinsic forceUpdate.

Having said that, I will look into implementing composition first, without breaking the current api.

I hope to have something available soon-ish.

**Comment by aksonov** [comment]
@jbrodriguez Sorry, i've found that reactive approach is much more elegant. Especially with libraries like Calmm https://github.com/calmm-js/documentation/blob/master/introduction-to-calmm.md

**Comment by jbrodriguez** [comment]
@aksonov hadn't heard about calmm-js. I will take a look.

Going functional is certainly more elegant, and there are libraries such as [Cyclejs](https://github.com/cyclejs) and [ffux](https://github.com/milankinen/ffux).

But it's certainly a paradigm shift and not everybody (thinking about companies, teams) is yet fully invested in it.
